include/zephyr/kernel/smp.h

```diff
@@ -28,4 +30,30 @@
  */
 void k_smp_cpu_start(int id);
 
+/**
+ * @brief Start one CPU with additional initialization callback.
```

Comment:
Similar comments as those for k_smp_cpu_start()

Reply:
This function is removed so this is no longer needed.